Kotlin/src/main/kotlin/com/gildedrose/updaters/ItemUpdater.kt

```diff
@@ -0,0 +1,21 @@
+package com.gildedrose.updaters
+import com.gildedrose.Item
+
+
+class ItemUpdater {
+
+    // update using corresponding function
+    fun update(item: Item) {
+        getUpdater(item).update(item)
+    }
+
+    private fun getUpdater(item: Item): BaseUpdater {
```

Comment:
Since this function just returns something with no additional logic, you can use a base expression for the body

```
private fun getUpdater(item: Item): BaseUpdater = when (item.name) {
    ItemType.conjured -> ConjuredUpdater()
    ItemType.backstagePass -> BackstagePassUpdater()
    ItemType.agedBrie -> AgedBrieUpdater()
    ItemType.legendary -> LegendaryUpdater()
    else -> NormalUpdater()
}
```